zebra-consensus/src/transaction.rs

```diff
@@ -112,34 +137,17 @@ where
             unimplemented!();
         }
 
-        let (tx, known_utxos, upgrade) = match req {
-            Request::Block {
-                transaction,
-                known_utxos,
-                height,
-            } => {
-                let upgrade = NetworkUpgrade::current(self.network, height);
-                (transaction, known_utxos, upgrade)
-            }
-            Request::Mempool {
-                transaction,
-                known_utxos,
-                upgrade,
-            } => (transaction, known_utxos, upgrade),
-        };
-
-        let mut spend_verifier = primitives::groth16::SPEND_VERIFIER.clone();
-        let mut output_verifier = primitives::groth16::OUTPUT_VERIFIER.clone();
-
-        let mut ed25519_verifier = primitives::ed25519::VERIFIER.clone();
-        let mut redjubjub_verifier = primitives::redjubjub::VERIFIER.clone();
-        let mut script_verifier = self.script_verifier.clone();
+        let script_verifier = self.script_verifier.clone();
+        let network = self.network;
 
+        let tx = req.transaction();
         let span = tracing::debug_span!("tx", hash = %tx.hash());
 
         async move {
+            let tx = req.transaction();
+
             tracing::trace!(?tx);
-            match &*tx {
+            match &**tx {
```

Comment:
Using [as_ref](https://doc.rust-lang.org/std/convert/trait.AsRef.html#tymethod.as_ref) shows the intent of the match.

It also avoids stacking up refs and derefs to get the exact match type you want, because Rust's [deref-coercion](https://doc.rust-lang.org/std/ops/trait.Deref.html#more-on-deref-coercion) will auto-deref as much as needed.

```suggestion
            match tx.as_ref() {
```

Reply:
Thanks! `.as_ref()` is a lot more readable :D